src/dodal/devices/eiger.py

```diff
@@ -34,27 +55,29 @@ def set(self, value, *, timeout=None, settle_time=None, **kwargs):
     stale_params = Component(EpicsSignalRO, "CAM:StaleParameters_RBV")
     bit_depth = Component(EpicsSignalRO, "CAM:BitDepthImage_RBV")
 
-    STALE_PARAMS_TIMEOUT = 60
-    GENERAL_STATUS_TIMEOUT = 10
-    # Long timeout for meta file to compensate for filesystem issues
-    META_FILE_READY_TIMEOUT = 30
-    ALL_FRAMES_TIMEOUT = 120
-    ARMING_TIMEOUT = 60
-
     filewriters_finished: StatusBase
 
     detector_params: DetectorParams | None = None
 
     arming_status = Status()
     arming_status.set_finished()
 
+    def __init__(self, beamline: str = "i03", *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.beamline = beamline
+        if beamline in AVAILABLE_TIMEOUTS:
+            self.timeouts = AVAILABLE_TIMEOUTS[beamline]
+        else:
+            raise ValueError(f"Unknown beamline_name: {beamline}")
```

Comment:
Could: I think it's better to use i03's timeout values as defaults rather than raising an error here. Then we could log a message to say that default values are being used

Reply:
Please could you also add a test for this line

Reply:
ok will do today